Sources/SQLKit/Query/SQLEnumType.swift

```diff
@@ -0,0 +1,39 @@
+/// `ENUM` type.
+///
+/// See `SQLDataType`.
+public struct SQLEnumType: SQLExpression {
+    /// The name of the enum type.
+    public var name: SQLExpression
+
+    /// The possible values of the enum type.
+    public var values: [SQLExpression]
+
+    /// Creates a new `SQLEnumType`.
+    public init(name: SQLExpression, values: [SQLExpression]) {
+        self.name = name
+        self.values = values
+    }
+
+    /// Creates a new `SQLEnumType`.
+    public init(name: String, values: [String]) {
+        self.init(name: SQLRaw(name), values: values.map { SQLLiteral.string($0) })
+    }
+
+    public func serialize(to serializer: inout SQLSerializer) {
+        switch serializer.dialect.enumSyntax {
+        case .inline(literal: let literal):
+            literal.serialize(to: &serializer)
+            SQLGroupExpression(values).serialize(to: &serializer)
+
+        case .typeName:
+            name.serialize(to: &serializer)
+
+        case .unsupported:
+            // NOTE: Consider using a CHECK constraint
+            //      with a TEXT type to verify that the
+            //      text value for a column is in a list
+            //      of possible options.
+            fatalError("ENUM types are unsupported by the current dialect.")
```

Comment:
I thought about this; it could even be taken a step further and turned into a `TEXT` field with a `CHECK` constraint that in combination really did semantically serve the purposes of `ENUM`. However, I think both of these options are a little too surprising. I would rather be told the enum type is not supported by the DB I am using (the fatal error option) and then be forced to pick another strategy explicitly.

One case you bring up above in a different comment that I was not thinking about when I made this PR is the case where the user is supporting more than one database type. In that case, it seems possible the user accidentally tests with a DB that allows enums and then ships a library or gets farther down the road when a different database hits the fatal error trap. That possibility alone is pushing me towards the opinion of just not implementing the enum type at the `SQLKit` layer. Realistically, the user is going to have to pick from dialects that support enums one way or the other (except if we give a fallback as you are suggesting) so it may as well only be explicitly supported by the dialects that allow it.

The motivation to put it here was to share the implementation amongst all of the many `MySQL`-like dialects, and that still feels valid, but there are definitely some substantive drawbacks you bring up.